src/main/kotlin/com/sourcegraph/cody/chat/AgentChatSession.kt

```diff
@@ -177,9 +183,41 @@ private constructor(
     if (messages.lastOrNull()?.id == message.id) {
       messages.removeLast()
     }
-    messages.add(message)
-    chatPanel.addOrUpdateMessage(message)
-    HistoryService.getInstance(project).updateChatMessages(internalId, messages)
+    if (messages.size == 0) {
+      if(chatPanel.modelDropdown.selectedItem != null) {
+        val selectedItem = chatPanel.modelDropdown.selectedItem
+        val displayModelName = (selectedItem as CodyModelComboboxItem).name
+        selectedModel = ChatModel.fromDisplayName(displayModelName)
+        sessionId.get().thenAccept { sessionId ->
+          CodyAgentService.applyAgentOnBackgroundThread(project) { agent ->
+            agent.server
+              .webviewReceiveMessage(
+                WebviewReceiveMessageParams(
+                  sessionId,
+                  WebviewMessage(command = "chatModel", model = selectedModel?.agentName)
+                )
+              )
+              .get()
+            messages.add(message)
+            chatPanel.addOrUpdateMessage(message)
+            ApplicationManager.getApplication().invokeLater {
+              HistoryService.getInstance(project)
+                .updateChatMessages(
+                  internalId, messages, selectedModel?.displayName ?: DEFAULT_MODEL.displayName
```

Comment:
That is kind of incorrect. If we don't know what model we selected, in case of `selectedModel` being null, shouldn't we also serialise it as a null?
And then when restored it will use the default again (and you must support this path anyway for historic reasons).